### DISCUSSION TITLE
draw overlay bitmap

### DISCUSSION THREAD

**Discussion by firemen01** [FactoryXCode/MfPack#48] [Q&A] [answered]
tell me please, it is very necessary to draw your own bitmap over the video, how can this be done?
draw not on the video frame itself, as the video may be of too low resolution, but on the form itself on top of.
maybe you can add OverlayBitmap (TBimap) property, and  TransparencyColor (TColor) and Rect (Output Rect on HWND) to the player?

Good luck!

**Comment by FactoryXCode** [comment, marked as answer]
Hi,
To do this you should examine the MfPlayer X sample. This sample has an overlay window where you can display/put any kind of control that is able to display an image or what ever you want on top of the video. The sample use this overlay to show video information and subtitles.

**Comment by firemen01** [comment]
thanks for the answer!
in your examples you just make the new shape transparent and set it on top of the video.
perhaps this example is suitable for imitating text or static images on top of a video, but I need to draw my own bitpum on each frame. is it possible somehow after drawing a video frame on top to draw another bitmap on the same canvas?
for example, to implement the output on the video background of the floating text of a karaoke song, when you need to draw the text at least 60 times per second to smoothly implement the movement of the text.
for example, in dspack, you can use a samplegrabber that allows you to read each video frame and draw it yourself further on any canvas, additionally draw something of your own on top.
Is there any other implementation possible in your library other than using a floatform or other HWND objects?

**Reply by FactoryXCode** [reply]
On the overlay you can put anything you want, even non static objects or even another video.
There're some pitfalls to consider: You have to implement exact calculations to follow the underlaying video size and position when needed, because the overlay will just follow the videowindow size and position.  
Basicly it's just a canvas independent from the underlaying video.
If you want to draw bitmaps or other images, I would suggest to place a TImage control on the overlay and calculate it's dimensions and position when needed.
